src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/LocalFolderUtility.cs

```diff
@@ -1076,5 +1076,18 @@ private static LocalPackageInfo GetPackageV3(string root, string id, string vers
 
             return null;
         }
+
+        private static FileInfo TryCreateFileInfo(string localPath)
+        {
+            FileInfo fileInfo = null;
+            try
+            {
+                fileInfo = new FileInfo(localPath);
+            }
+            catch (System.IO.PathTooLongException)
```

Comment:
Consider handling other exception types it may throw:

|Exception|	Condition|
| --- | --- |
|SecurityException	| The caller does not have the required permission. |
|ArgumentException	|The file name is empty, contains only white spaces, or contains invalid characters.|
|UnauthorizedAccessException	|Access to fileName is denied.|
|NotSupportedException	|fileName contains a colon (:) in the middle of the string.|